commonv5/ts/__tests__/publish-task-test.ts

```diff
@@ -317,3 +341,122 @@ it("task should not fail the task even if all ceTasks timeout", async () => {
+      status: "status",
+      type: EndpointType.SonarQube,
+      componentName: "componentName",
+      warnings: [],
+    });
+
+    // Mock finding the report file to process
+    jest.spyOn(TaskReport, "createTaskReportsFromFiles").mockResolvedValue([TASK_REPORT]);
+    jest.spyOn(Task, "waitForTaskCompletion").mockResolvedValue(returnedTask);
+
+    jest.spyOn(apiUtils, "addBuildProperty").mockResolvedValue(null);
+    jest.spyOn(apiUtils, "getAuthToken").mockImplementation(() => null);
+    jest.spyOn(serverUtils, "fillBuildProperty").mockImplementation(() => null);
+    jest.spyOn(serverUtils, "publishBuildSummary").mockImplementation(() => null);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("not fail when measures can not be retrieved", async () => {
+    tl.setVariable(TaskVariables.SonarQubeEndpoint, SQ_ENDPOINT.toJson());
+    jest.spyOn(api, "fetchProjectStatus").mockResolvedValueOnce(PROJECT_STATUS_OK);
+
+    await publishTask.default(EndpointType.SonarQube);
+
+    expect(tl.debug).toHaveBeenCalledWith(
+      "Unable to get measures. It is expected if you are not using a user token but instead a global or project analysis token.",
+    );
+  });
+
+  it("should show available fetched measures", async () => {
+    tl.setVariable(TaskVariables.SonarQubeEndpoint, SQ_ENDPOINT.toJson());
+    jest.spyOn(api, "fetchProjectStatus").mockResolvedValueOnce(PROJECT_STATUS_OK);
+    jest.spyOn(api, "fetchComponentMeasures").mockResolvedValueOnce([
+      {
+        metric: "new_violations",
+        value: "10",
+      },
+    ]);
+
+    await publishTask.default(EndpointType.SonarQube);
+
+    // spy on publishBuildSummary
+    expect(serverUtils.publishBuildSummary).toHaveBeenCalledTimes(1);
+    const buildSummary = (serverUtils.publishBuildSummary as any).mock.calls[0][0];
+    expect(buildSummary).toMatch("Quality Gate passed (componentName)");
+    expect(buildSummary).toMatch("10 new issues");
+    expect(buildSummary).toMatch("100% Coverage on new code");
+  });
+
+  it("should not fail when no measure/metric is available", async () => {
+    tl.setVariable(TaskVariables.SonarQubeEndpoint, SQ_ENDPOINT.toJson());
+    jest
+      .spyOn(api, "fetchProjectStatus")
+      .mockResolvedValueOnce({ ...PROJECT_STATUS_OK, conditions: [] });
+    jest.spyOn(api, "fetchComponentMeasures").mockResolvedValueOnce([]);
+
+    await publishTask.default(EndpointType.SonarQube);
+
+    // spy on publishBuildSummary
+    expect(serverUtils.publishBuildSummary).toHaveBeenCalledTimes(1);
+    const buildSummary = (serverUtils.publishBuildSummary as any).mock.calls[0][0];
+    expect(buildSummary).toMatch("Quality Gate passed (componentName)");
+  });
+
+  it.each([
+    [EndpointType.SonarQube, SQ_ENDPOINT, { "sonar.pullrequest.key": "123" }, true],
+    [EndpointType.SonarQube, SQ_ENDPOINT, {}, false],
+    [EndpointType.SonarQube, SQ_ENDPOINT, { "sonar.branch.name": "some-branch" }, false],
```

Comment:
Are these 2 cases relevant? can we show fixed issues if there is no pr key?

Reply:
You're right, if there's no PR key, the measure can't be fetched on SQ anyway, so I've dropped it, thanks